reference/test/factorization/ic_kernels.cpp

```diff
@@ -0,0 +1,239 @@
+
+    void apply_impl(const gko::LinOp *alpha, const gko::LinOp *b,
+                    const gko::LinOp *beta, gko::LinOp *x) const override
+    {}
+};
+
+
+template <typename ValueIndexType>
+class Ic : public ::testing::Test {
+protected:
+    using value_type =
+        typename std::tuple_element<0, decltype(ValueIndexType())>::type;
+    using index_type =
+        typename std::tuple_element<1, decltype(ValueIndexType())>::type;
+    using factorization_type = gko::factorization::Ic<value_type, index_type>;
+    using Coo = gko::matrix::Coo<value_type, index_type>;
+    using Csr = gko::matrix::Csr<value_type, index_type>;
+    using Dense = gko::matrix::Dense<value_type>;
+
+    Ic()
+        : ref(gko::ReferenceExecutor::create()),
+          exec(std::static_pointer_cast<const gko::Executor>(ref)),
+          identity(gko::initialize<Csr>(
+              {{1., 0., 0.}, {0., 1., 0.}, {0., 0., 1.}}, ref)),
+          banded(gko::initialize<Csr>(
+              {{1., 1., 0.}, {1., 2., 1.}, {0., 1., 2.}}, ref)),
+          banded_l_expect(gko::initialize<Csr>(
+              {{1., 0., 0.}, {1., 1., 0.}, {0., 1., 1.}}, ref)),
+          mtx_system(gko::initialize<Csr>({{9., 0., -6., 3.},
+                                           {0., 36., 18., 24.},
+                                           {-6., 18., 17., 14.},
+                                           {-3., 24., 14., 18.}},
+                                          ref)),
+          mtx_l_system(gko::initialize<Csr>({{9., 0., 0., 0.},
+                                             {0., 36., 0., 0.},
+                                             {-6., 18., 17., 0.},
+                                             {-3., 24., 14., 18.}},
+                                            ref)),
+          mtx_l_system_coo(Coo::create(exec)),
+          mtx_l_init_expect(gko::initialize<Csr>(
+              {{3., 0., 0., 0.},
+               {0., 6., 0., 0.},
+               {-6., 18., static_cast<value_type>(sqrt(17.)), 0.},
+               {-3., 24., 14., static_cast<value_type>(sqrt(18.))}},
+              ref)),
+          mtx_l_it_expect(gko::initialize<Csr>({{3., 0., 0., 0.},
+                                                {0., 6., 0., 0.},
+                                                {-2., 3., 2., 0.},
+                                                {-1., 4., 0., 1.}},
+                                               ref)),
+          fact_fact(factorization_type::build().on(exec)),
+          tol{r<value_type>::value}
+    {
+        mtx_l_system->convert_to(gko::lend(mtx_l_system_coo));
+    }
+
+    std::shared_ptr<const gko::ReferenceExecutor> ref;
+    std::shared_ptr<const gko::Executor> exec;
+    std::shared_ptr<Csr> identity;
+    std::shared_ptr<Csr> banded;
+    std::shared_ptr<Csr> banded_l_expect;
+    std::shared_ptr<Csr> mtx_system;
+    std::unique_ptr<Csr> mtx_l_system;
+    std::unique_ptr<Coo> mtx_l_system_coo;
+    std::unique_ptr<Csr> mtx_l_init_expect;
```

Comment:
This variable is not used, is there a test missing or can this be removed?

Reply:
Good catch, those are left-over from ParIC tests.